build.gradle

```diff
@@ -12,19 +12,19 @@ plugins {
     id 'application'
     id 'checkstyle'
     id 'jacoco'
-    id 'com.github.ben-manes.versions' version '0.12.0'
-    id 'com.github.johnrengelman.shadow' version '1.2.3'
+    id 'com.github.ben-manes.versions' version '0.17.0'
+    id 'com.github.johnrengelman.shadow' version '2.0.2'
     id 'com.install4j.gradle' version '7.0.1'
-    id 'de.undercouch.download' version '3.2.0'
-    id 'net.ltgt.errorprone' version '0.0.11'
+    id 'de.undercouch.download' version '3.3.0'
+    id 'net.ltgt.errorprone' version '0.0.13'
```

Comment:
Note that this is triggering a lot of new warnings.  Most of them look legitimate.  However, one in particular, ClassCanBeStatic, is triggering a lot of false negatives in test code.  The JUnit `@Nested` annotation requires nested tests to be inner classes (non-static) rather than nested classes (static), so we have no choice here.

Does Error Prone offer an annotation mechanism to ignore individual warnings?  Or is there a way to have Error Prone ignore this specific warning for all classes that have a `Test` suffix?

Reply:
@ssoloff The latest version offers a way to exclude classes with filenames that match a given regular expression.
I disabled error prone for all files inside src/test and src/integ_test folder.
There is a legacy way of doing it that by annotating classes with `@SuppressWarnings("<errorname>")`, in this case : `@SuppressWarnings("ClassCanBeStatic")`, but it's really not pretty, especially because custom compilers like eclipse will warn you of this uneccessary annotation